acs5racedemographics.py

```diff
@@ -0,0 +1,182 @@
+                'race_hispaniclatino_asian', 'B03002_015E': \
+                'race_hispaniclatino_native', 
+                'B03002_014E': 'race_hispaniclatino_black',\
+                'B03002_018E': 'race_hispaniclatino_other', \
+                'B03002_019E': 'race_hispaniclatino_twoplus_total'}
+
+    race_ls = list(var_dict.keys())
+    value_ls = list(var_dict.values())
+                    
+    detailed_table = 'https://api.census.gov/data/2018/acs/acs5?'
+
+    response = getCensusResponse(detailed_table, race_ls, geo)
+    
+    #race_data is list of lists, where list elements are rows
+    #headers defined by group_data_variables elements
+    race_data = response.json()
+    
+    #Labels table columns with group variable labels
+    labelled_race_ls = [[]]
+
+    for i, row in enumerate(race_data):
+        #header row
+        if i == 0:
+            for c, col in enumerate(race_data[0]):
+                #Gets variable label from group data
+                if col in var_dict:
+                    label = var_dict[col]
+                    labelled_race_ls[0].append(label)
+                #Variables not in group data added to list
+                else:
+                    labelled_race_ls[0].append(col)
+        #converts data strings to ints
+        else:
+            new_row = []
+            for v, n in enumerate(row):
+                if v == 0 or v == len(row)-1:
+                    new_row.append(n)
+                else:
+                    new_row.append(int(n))
+
+            labelled_race_ls.append(new_row)
+
+    race_data_IL = []
+    
+    if "zip" == geography:
+        #filter zipcodes
+        #IL zipcodes begin with numbers in zip_ls_IL
+        zip_ls_IL = ['60', '61', '62']
+
+        for d in labelled_race_ls[1:]:
+            #print(d[3][:2])
+            if d[-1][:2] in zip_ls_IL:
+                race_data_IL.append(d)
+    else:
+        #creates county geo labels
+        race_data_IL = [d[:-2] + [f"{d[-2]}{d[-1]}"] for d in labelled_race_ls[1:]]
+            
+    #Json format
+    #{'geography':
+        #{'metric_one': 1234,
+        #'metric_two': 5678}}
+        
+    final_json = {}
+
+    for d in race_data_IL:
+        #set variables to list elements
+        #determine whether to pull unused variables
+        name, race_hispaniclatino_twoplus_exclusive, \
+        race_hispaniclatino_twoplus_inclusive, race_total, race_native, race_black,\
+        race_white, race_total_not_hispaniclatino, race_twoplus_total, race_pacific,\
+        race_other, race_asian, race_hispaniclatino_white, race_hispaniclatino_total,\
+        race_twoplus_exclusive, race_twoplus_inclusive, race_hispaniclatino_pacific,\
+        race_hispaniclatino_asian, race_hispaniclatino_native, \
+        race_hispaniclatino_black, race_hispaniclatino_other, \
+        race_hispaniclatino_twoplus_total, g = d
+       
+        #Calculate percentages?
```

Comment:
Wonder how hard it would be to do this? I'm thinking the client probably would want to see the percentage instead of the number of people that were sampled?

Reply:
Not sure if it makes more sense to calculate it here (and add it to the json) or calculate it on the front end.

Reply:
Good question... My gut feeling is that the backend should calculate it, but lets discuss

Reply:
Are we going to put the data into a database or just send them through as JSONs?

If we are putting them into a database, I would query the percent there instead of here.

Reply:
I think we want to calculate it here so that the FE doesn't have to. Firebase will just store the json as a string I believe (its a NOSQL type deal)